### PR TITLE
Add Component searching to the SceneDebugger

### PR DESCRIPTION
**Pull request by killzoms**
Note needs exact name

### CODE REVIEW

**Comment by killzoms** [comment]
Would do a typeName contains search but im afraid it may return to many gameObjects and affect performance

File: NitroxClient/Debuggers/SceneDebugger.cs [hunk 1]
```diff
@@ -256,6 +256,8 @@ private void RenderTabHierarchy()
                                 }
                             }
 
+                            Log.Info("Primary");
```
Comment (Measurity) [inline NitroxClient/Debuggers/SceneDebugger.cs]: Logging still useful?

File: NitroxClient/Debuggers/SceneDebugger.cs [hunk 2]
```diff
@@ -293,7 +295,18 @@ private void RenderTabHierarchy()
 
                     if (string.IsNullOrEmpty(gameObjectSearchPatternInvalidMessage))
                     {
+                        Type type = AppDomain.CurrentDomain.GetAssemblies()
+                        .Select(a => a.GetType(gameObjectSearch))
```
Comment (Measurity) [inline NitroxClient/Debuggers/SceneDebugger.cs]: Case insensitive search might be good here so you don't have to type the name exactly.

File: NitroxClient/Debuggers/SceneDebugger.cs [hunk 3]
```diff
@@ -305,17 +318,20 @@ private void RenderTabHierarchy()
                 using (GUILayout.ScrollViewScope scroll = new GUILayout.ScrollViewScope(hierarchyScrollPos))
                 {
                     hierarchyScrollPos = scroll.scrollPosition;
-
+                    Log.Info("Secondary");
```
Comment (Measurity) [inline NitroxClient/Debuggers/SceneDebugger.cs]: Logging still useful?

**Comment by killzoms** [comment]
image

**Pull request merged**